### PR TITLE
added e2e test for CloudBuildSource

### PR DESCRIPTION
**Pull request by capri-xiyue**
Fixes #

## Proposed Changes

- 🎁 added smoke test for CloudBuildSource
- 🎁 added TestCloudBuildSourceWithTarget

### CODE REVIEW

File: test/e2e/lib/build.go [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+	}
+	build := new(cloudbuild.Build)
+	image := CloudBuildImage(project, imageName)
+	build.Steps = append(build.Steps, &cloudbuild.BuildStep{
+		Name: "gcr.io/cloud-builders/docker",
+		Args: []string{"build", "-t", image, "."},
+	})
+	build.Images = []string{image}
+	projectsBuildsCreateCall := cloudbuildService.Projects.Builds.Create(project, build)
+	operation, err := projectsBuildsCreateCall.Do()
```
Comment (ian-mi) [inline test/e2e/lib/build.go:86]: I would combine these lines

File: test/e2e/lib/build.go [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+	"errors"
+	"fmt"
+	"os"
+	"testing"
+
+	v1 "k8s.io/api/core/v1"
+
+	kngcptesting "github.com/google/knative-gcp/pkg/reconciler/testing"
+	"github.com/google/knative-gcp/test/e2e/lib/resources"
+	"google.golang.org/api/cloudbuild/v1"
```
Comment (ian-mi) [inline test/e2e/lib/build.go:31]: I would prefer the cloud.google.com/go/cloudbuild/apiv1/v2 client

File: test/e2e/lib/deletion.go [hunk 3]
```diff
@@ -29,6 +29,15 @@ func (c *Client) DeletePubSubOrFail(name string) {
 	}
 }
 
+func (c *Client) DeleteBuildOrFail(name string) {
+	c.T.Helper()
+	builds := c.KnativeGCP.EventsV1alpha1().CloudBuildSources(c.Namespace)
```
Comment (ian-mi) [inline test/e2e/lib/deletion.go:34]: Shouldn't this be V1beta1?

File: test/e2e/lib/build.go (hunk 2 above)
Comment (capri-xiyue) [reply test/e2e/lib/build.go:31]: There is no v2 client https://godoc.org/google.golang.org/api/cloudbuild

Comment (ian-mi) [reply test/e2e/lib/build.go:31]: That's a different package: https://pkg.go.dev/cloud.google.com/go/cloudbuild/apiv1/v2?tab=doc.

File: test/e2e/lib/get.go [hunk 4]
```diff
@@ -32,6 +32,16 @@ func (c *Client) GetPubSubOrFail(name string) *eventsv1alpha1.CloudPubSubSource
 	return existing
 }
 
+func (c *Client) GetBuildOrFail(name string) *eventsv1alpha1.CloudBuildSource {
+	c.T.Helper()
+	builds := c.KnativeGCP.EventsV1alpha1().CloudBuildSources(c.Namespace)
```
Comment (ian-mi) [inline test/e2e/lib/get.go:37]: v1beta1 here as well?

Comment (capri-xiyue) [reply test/e2e/lib/get.go:37]: done

File: test/e2e/lib/build.go (hunk 2 above)
Comment (capri-xiyue) [reply test/e2e/lib/build.go]: done

**Pull request merged**